### PR TITLE
S tuli/stateless

### PR DESCRIPTION
**Pull request by s-tuli**
Converting the DataX.Metrics stateful ServiceFabric app to be stateless ServiceFabric app. Tested it by deploying to a test cluster. Unit tests are passing.

### CODE REVIEW

File: Services/DataX.Metrics/DataX.Metrics/ApplicationParameters/Local.1Node.xml [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
-  </Parameters>
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!--
+Copyright (c) Microsoft Corporation.  All rights reserved.
+Licensed under the MIT License
+-->
+<Application xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="fabric:/DataX.Metrics" xmlns="http://schemas.microsoft.com/2011/01/fabric">
+  <Parameters>
+    <Parameter Name="DataX.Metrics.Ingestor_ASPNETCORE_ENVIRONMENT" Value="Development" />
+    <Parameter Name="DataX.Metrics.Ingestor_InstanceCount" Value="-1" />
```
Comment (kjcho-msft) [inline Services/DataX.Metrics/DataX.Metrics/ApplicationParameters/Local.1Node.xml]: for local, just use 1.

File: Services/DataX.Metrics/DataX.Metrics/ApplicationParameters/Local.5Node.xml [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
-  </Parameters>
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!--
+Copyright (c) Microsoft Corporation.  All rights reserved.
+Licensed under the MIT License
+-->
+<Application xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="fabric:/DataX.Metrics" xmlns="http://schemas.microsoft.com/2011/01/fabric">
+  <Parameters>
+    <Parameter Name="DataX.Metrics.Ingestor_ASPNETCORE_ENVIRONMENT" Value="Development" />
+    <Parameter Name="DataX.Metrics.Ingestor_InstanceCount" Value="-1" />
```
Comment (kjcho-msft) [inline Services/DataX.Metrics/DataX.Metrics/ApplicationParameters/Local.5Node.xml]: same here. let's use 1

File: Services/DataX.Metrics/DataX.Metrics.Ingestor/Helper/IngestorEventProcessor.cs [hunk 3]
```diff
@@ -71,57 +71,63 @@ public Task ProcessErrorAsync(PartitionContext context, Exception error)
...
                         {"MessagesRecieved", messageCount},
                         {"MetricsSent", metricCount},
-                        {"ElapsedTimeInMs",  sw.ElapsedMilliseconds}
-                });
+                        {"ElapsedTimeInMs",  sw.ElapsedMilliseconds}
+                    });
+            }
+            catch(Exception e)
+            {
+                _logger.LogError(e, "Exception thrown in ProcessEventsAsync");
```
Comment (kjcho-msft) [inline Services/DataX.Metrics/DataX.Metrics.Ingestor/Helper/IngestorEventProcessor.cs]: log e.Message as well.

File: Services/DataX.Metrics/DataX.Metrics/ApplicationParameters/Local.1Node.xml (hunk 1 above)
Comment (s-tuli) [reply Services/DataX.Metrics/DataX.Metrics/ApplicationParameters/Local.1Node.xml]: Updated

File: Services/DataX.Metrics/DataX.Metrics.Ingestor/Helper/IngestorEventProcessor.cs (hunk 3 above)
Comment (s-tuli) [reply Services/DataX.Metrics/DataX.Metrics.Ingestor/Helper/IngestorEventProcessor.cs]: Updated

**Pull request merged**